src/sentry/eventstore/models.py

```diff
@@ -552,6 +552,25 @@ def _get_column_name(self, column: Columns) -> str:
         # Events are currently populated from the Events dataset
         return cast(str, column.value.event_name)
 
+    @property
+    def has_stacktrace(self) -> bool:
+        """
+        Whether or not this event has a stacktrace.
+
+        Note that a stacktrace must have at least one frame or it doesn't count.
+        """
+        has_top_level_frames = bool(get_path(self.data, "stacktrace", "frames"))
+
+        exceptions = get_path(self.data, "exception", "values") or []
+        has_exception_frames = any(
+            get_path(exception, "stacktrace", "frames") for exception in exceptions
+        )
+
+        threads = get_path(self.data, "threads", "values") or []
+        has_thread_frames = any(get_path(thread, "stacktrace", "frames") for thread in threads)
```

Comment:
Do you have an example of where stack traces are extracted from threads?
So far, I have not seen stack traces being used to group mobile issues.

Reply:
There are a number of snapshot inputs where that's the case:

https://github.com/getsentry/sentry/blob/master/tests/sentry/grouping/grouping_inputs/cocoa_dispatch_client_callout.json
https://github.com/getsentry/sentry/blob/master/tests/sentry/grouping/grouping_inputs/block_invoke.json
https://github.com/getsentry/sentry/blob/master/tests/sentry/grouping/grouping_inputs/threads-compute-hashes.json

Also, it's part of the event schema: 
https://develop.sentry.dev/sdk/data-model/event-payloads/threads/
https://github.com/getsentry/sentry-data-schemas/blob/main/relay/event.schema.json#L3844